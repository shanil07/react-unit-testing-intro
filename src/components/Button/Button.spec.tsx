import { render } from "@testing-library/react";
import { Button } from "./index";

describe("components/Button", () => {
  it("should render component", () => {
    const { getByRole } = render(<Button>Test text</Button>);
    expect(getByRole("button")).toHaveTextContent("Test text");
  });
  it("should match snapshot", () => {
    const { container } = render(<Button>Test text</Button>);
    expect(container).toMatchSnapshot();
  });
});
