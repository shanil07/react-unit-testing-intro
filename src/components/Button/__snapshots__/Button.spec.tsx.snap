// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button should match snapshot 1`] = `
<div>
  <button
    class="default"
  >
    Test text
  </button>
</div>
`;
